runtime-transaction/src/runtime_transaction/transaction_view.rs

```diff
@@ -91,6 +100,47 @@ impl<D: TransactionData> RuntimeTransaction<ResolvedTransactionView<D>> {
     }
 }
 
+impl<D: TransactionData> TransactionWithMeta for RuntimeTransaction<ResolvedTransactionView<D>> {
+    fn as_sanitized_transaction(&self) -> impl Borrow<SanitizedTransaction> {
+        let VersionedTransaction {
+            signatures,
+            message,
+        } = self.to_versioned_transaction();
+
+        let is_writable_account_cache = (0..self.transaction.total_num_accounts())
+            .map(|index| self.is_writable(usize::from(index)))
+            .collect();
+
+        let message = match message {
+            VersionedMessage::Legacy(message) => SanitizedMessage::Legacy(LegacyMessage {
+                message: Cow::Owned(message),
+                is_writable_account_cache,
+            }),
+            VersionedMessage::V0(message) => SanitizedMessage::V0(LoadedMessage {
+                message: Cow::Owned(message),
+                loaded_addresses: Cow::Owned(self.loaded_addresses().unwrap().clone()),
+                is_writable_account_cache,
+            }),
+        };
+
+        // SAFETY:
+        // - Simple conversion between different formats
+        // - `ResolvedTransactionView` has undergone sanitization checks
+        unsafe {
+            SanitizedTransaction::new_from_parts(
+                message,
+                *self.message_hash(),
+                self.is_simple_vote_transaction(),
+                signatures,
+            )
+        }
+    }
+
+    fn to_versioned_transaction(&self) -> VersionedTransaction {
+        bincode::deserialize(self.transaction.data()).unwrap()
```

Comment:
We already have all the computed data frames in the resolved view, wouldn't it be better to use those to construct the versioned transaction rather than doing deserialization which basically has to recompute them? Can remove the `bincode` dependency then too